checker.cpp

```diff
@@ -1,22 +1,20 @@
-#include <assert.h>
-#include <iostream>
-using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7) == true);
-  assert(batteryIsOk(50, 85, 0) == false);
-}
+#include"checker.h"
+
+int main() {
+  checker checkerObj(new BatteryMgmtSystmTest(25, 70, 0.7));

```

Comment:
Try to cover boundary conditions, out of range values and in range values, for example test values for temperature could be -1, 0, 25, 45, 50. Same can be applied to other parameters too.